packages/siopv2-oid4vp-rp-auth/src/functions.ts

```diff
@@ -118,39 +134,127 @@ export async function createRPBuilder(args: {
+    )
     .withRevocationVerification(RevocationVerification.NEVER)
     .withPresentationVerification(getPresentationVerificationCallback(didOpts.idOpts, context))
-
-  if (!rpOpts.clientMetadataOpts?.subjectTypesSupported) {
+  if (hasher) {
+    builder.withHasher(hasher)
+  }
+  //fixme: this has been removed in the new version of did-auth-siop
+  /*if (!rpOpts.clientMetadataOpts?.subjectTypesSupported) {
     // Do not update in case it is already provided via client metadata opts
     didMethods.forEach((method) => builder.addDidMethod(method))
-  }
-  builder.withWellknownDIDVerifyCallback(getWellKnownDIDVerifyCallback(didOpts, context))
+  }*/
+  //fixme: this has been removed in the new version of did-auth-siop
+  // builder.withWellknownDIDVerifyCallback(getWellKnownDIDVerifyCallback(didOpts, context))
 
   if (definition) {
     builder.withPresentationDefinition({ definition }, PropertyTarget.REQUEST_OBJECT)
   }
 
-  const key = resolution.key
+  //const key = resolution.key
+  //fixme: this has been removed in the new version of did-auth-siop
+  //builder.withSuppliedSignature(SuppliedSigner(key, context, getSigningAlgo(key.type) as unknown as KeyAlgo), did, kid, getSigningAlgo(key.type))
 
-  if (isManagedIdentifierDidResult(resolution)) {
+  /*if (isManagedIdentifierDidResult(resolution)) {
     //fixme: only accepts dids in version used. New SIOP lib also accepts other types
     builder.withSuppliedSignature(
       SuppliedSigner(key, context, getSigningAlgo(key.type) as unknown as KeyAlgo),
       resolution.did,
       resolution.kid,
       getSigningAlgo(key.type),
     )
-  }
+  }*/
+  //fixme: signcallback and it's return type are not totally compatible with our CreateJwtCallbackBase
+  const createJwtCallback = signCallback(rpOpts.didOpts.idOpts, context)
+  builder.withCreateJwtCallback(createJwtCallback as unknown as CreateJwtCallback<any>)
   return builder
 }
 
+//fixme: this is written based on OID4VCIHolder.signCallback sync the fixes to this with that function
+export function signCallback(
+  idOpts: ManagedIdentifierOpts,
+  context: IRequiredContext,
+): (jwt: { header: JwtHeader; payload: JwtPayload }, kid?: string) => Promise<string> {
+  return async (jwt: { header: JwtHeader; payload: JwtPayload }, kid?: string) => {
+    let iss = jwt.payload.iss
+    const jwk = jwt.header.jwk
+
+    if (!kid) {
+      kid = jwt.header.kid
+    }
+    if (!kid) {
+      kid = idOpts.kid
+    }
+    if (!kid && jwk && 'kid' in jwk) {
+      kid = jwk.kid as string
+    }
+
+    if (kid && !idOpts.kid) {
+      // sync back to id opts
+      idOpts.kid = kid.split('#')[0]
+    }
+
+    const resolution = await context.agent.identifierManagedGet(idOpts)
+    if (!iss && isManagedIdentifierDidResult(resolution)) {
+      iss = resolution.did
+    } else {
+      iss = resolution.issuer
+    }
+    if (!iss) {
+      return Promise.reject(Error(`No issuer could be determined from the JWT ${JSON.stringify(jwt)}`))
```

Comment:
See remark for the other signer. Please just call the compact jwt signer. Most of the rest is not needed here

Reply:
refactored the function to use `jwtCreateJwsCompactSignature` function